cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MdsHistoryKey.java

```diff
@@ -47,30 +53,30 @@ static long extractTime(byte[] rowKey, String type) {
     keySplitter.skipBytes();
 
     switch (type) {
-      case MetadataEntity.PROGRAM:
+      case PROGRAM:
         keySplitter.skipString();
         keySplitter.skipString();
         keySplitter.skipString();
         keySplitter.skipString();
         break;
-      case MetadataEntity.APPLICATION:
+      case APPLICATION:
```

Comment:
you can still avoid repetition by using multiple case labels. Please see my original comment.